calc/vxlan_resolver.go

```diff
@@ -159,66 +182,110 @@ func (c *VXLANResolver) OnBlockUpdate(update api.Update) (_ bool) {
 	return
 }
 
+func (c *VXLANResolver) OnResourceUpdate(update api.Update) (_ bool) {
+	resourceKey := update.Key.(model.ResourceKey)
+	if resourceKey.Kind != "node" {
+		return
+	}
+
+	nodeName := update.Key.(model.ResourceKey).Name
+	logCxt := logrus.WithField("node", nodeName)
+
+	if update.Value != nil {
+		node := update.Value.(*apiv3.Node)
+
+		bgp := node.Spec.BGP
+		if bgp != nil {
+			c.nodeNameToNode[nodeName] = node
+
+			ipv4, _, err := cnet.ParseCIDROrIP(bgp.IPv4Address)
+			if err != nil {
+				logCxt.Error(err.Error())
```

Comment:
Let's use the `WithError` functionality here, and add a contextual message.

e.g. `logCxt.WithError(error).Error("Failed to parse IPv4 BGP address")`